solidity/contracts/deposit/DepositFunding.sol

```diff
@@ -168,84 +160,10 @@ library DepositFunding {
         bool _isFraud = _d.submitSignatureFraud(_v, _r, _s, _signedDigest, _preimage);
         require(_isFraud, "Signature is not fraudulent");
         _d.logFraudDuringSetup();
-
-        // If the funding timeout has elapsed, punish the funder too!
-        if (block.timestamp > _d.fundingProofTimerStart.add(TBTCConstants.getFundingTimeout())) {
-            _d.setFailedSetup();
-        } else {
-            /* NB: This is reuse of the variable */
-            _d.fundingProofTimerStart = block.timestamp;
-            _d.setFraudAwaitingBTCFundingProof();
-        }
-    }
-
-    /// @notice     Anyone may notify the contract no funding proof was submitted during funding fraud.
-    /// @dev        This is not a funder fault. The signers have faulted, so the funder shouldn't fund.
-    /// @param  _d  Deposit storage pointer.
-    function notifyFraudFundingTimeout(DepositUtils.Deposit storage _d) public {
-        require(
-            _d.inFraudAwaitingBTCFundingProof(),
-            "Not currently awaiting fraud-related funding proof"
-        );
-        require(
-            block.timestamp > _d.fundingProofTimerStart.add(TBTCConstants.getFraudFundingTimeout()),
-            "Fraud funding proof timeout has not elapsed"
-        );
-        _d.setFailedSetup();
-        _d.logSetupFailed();
-
-        partiallySlashForFraudInFunding(_d);
+        distributeSignerBondsToFunder(_d);
```

Comment:
Note: funder can cause this to fail and potentially push a fraudulent deposit into an active state.

Reply:
Indeed, this feeds into the overall “push vs pull” question noted in a few other places as well:
- https://github.com/keep-network/tbtc/pull/524#discussion_r392771211 (liquidation)
- https://github.com/keep-network/tbtc/pull/520#discussion_r399326416 (also funding, but a different scenario)
- I believe a couple of other places we haven't called out explicitly but that we're taking stock of at the moment.